numerics/polynomial.hpp

```diff
@@ -0,0 +1,103 @@
+
+#pragma once
+
+#include <tuple>
+#include <utility>
+
+#include "quantities/named_quantities.hpp"
+
+namespace principia {
+namespace numerics {
+namespace internal_polynomial {
+
+using quantities::Derivative;
+
+// TODO(phl): We would like to define NthDerivative in named_quantities.hpp
+// thus:
+//
+//   template<typename Value, typename Argument, int order>
+//   using NthDerivative = typename std::conditional_t<
+//       order == 0,
+//       Value,
+//       Quotient<Difference<Value>,
+//                Exponentiation<Difference<Argument>, order>>>;
+//
+//   template<typename Value, typename Argument>
+//   using Derivative = NthDerivative<Value, Argument, 1>;
+//
+// Unfortunately VS2015 is buggy and this interacts poorly with the
+// std::integer_sequence below (we get the wrong types).  Revisit once MSFT has
+// fixed their bugs.
```

Comment:
I'm not sure this is going to happen unless we make a bug report, considering that https://developercommunity.visualstudio.com/content/problem/106570/incorrect-compilation-error-on-c-code-involving-te.html only kicked the bug down the road to https://developercommunity.visualstudio.com/content/problem/165535/incorrect-compilation-error-on-c-operators-and-dec.html.

Maybe we should consider switching to another compiler front-end...

Reply:
I guess I should create a bug with MSFT.